src/contrib/persistence/Akka.Persistence.Sql.Common/Journal/QueryExecutor.cs

```diff
@@ -628,8 +628,8 @@ protected DbCommand GetCommand(DbConnection connection, string sql)
         protected virtual void WriteEvent(DbCommand command, IPersistentRepresentation e, IImmutableSet<string> tags)
         {
             var manifest = string.IsNullOrEmpty(e.Manifest) ? e.Payload.GetType().TypeQualifiedName() : e.Manifest;
-            var serializer = Serialization.FindSerializerFor(e.Payload);
-            var binary = serializer.ToBinary(e.Payload);
+            var serializer = Serialization.FindSerializerForType(typeof(IPersistentRepresentation));
```

Comment:
The SQL columns should only be used for storing fields that need to be queried on.  This is the pattern that JVM Akka follows for their Akka plugins.  All of their supported plugins just use the PersistentRepr protobuf serializer, mainly because it does the heavy lifting of ensuring that the actual payload is properly handled.